hello-world-master/source/components/scene/helloworld.brs

```diff
@@ -1,36 +1,35 @@
 sub init()
     m.imageRowList = m.top.findNode("imageRowList")
     m.button = m.top.findNode("button")
-    showImage()
     m.imageRowList.setFocus(true)
     m.button.observeField("buttonSelected", "onButtonSelected")
     m.imageRowList.observeField("rowItemSelected", "onRowItemSelected")
+    m.getRequestTask = CreateObject("roSGNode", "GetRequestTask")
+    m.getRequestTask.ObserveField("itemContent", "onFetchData")
+	  m.getRequestTask.control = "RUN"
+    m.newScreen = CreateObject("roSGNode", "DescriptionScreen")
+    m.newScreen.visible = false
+    m.top.appendChild(m.newScreen)
 end sub
 
-sub showImage()
-    listRoot = CreateObject("roSGNode","ContentNode")
-    row = CreateObject("roSGNode","ContentNode")
-    for i = 1 to 5 
-        rowChild = CreateObject("roSGNode","ContentNode")
-        rowChild.FHDPosterUrl = Substitute("pkg:/images/item{0}.jpg",i.toStr())
-        row.appendChild(rowChild)
-    end for
-    
-    listRoot.appendChild(row)
-    m.imageRowList.content = listRoot
+
+sub onFetchData() as void
+  m.imageRowList.content =  m.getRequestTask.itemcontent
 end sub
 
+
 function onKeyEvent(key as String, press as Boolean) as Boolean
     handled = false
     if press then
       if (key = "down" and m.imageRowList.hasFocus()) then
         handled = true
         m.button.setFocus(true)
-      else 
-        if (key = "up" and m.button.hasFocus()) then
+      else if (key = "up" and m.button.hasFocus()) then
             handled = true
             m.imageRowList.setFocus(true)
-        end if
+      else if (key ="back") then
+        m.newScreen.visible = false
+        handled = true
```

Comment:
Considering my previous suggestion with navigating to the `DescriptionScreen`, I think we could move the handle of key back in the DescriptionScreen and instead of making the screen invisible, we could remove it - https://developer.roku.com/en-ca/docs/references/brightscript/interfaces/ifsgnodechildren.md#removechildchild-as-rosgnode-as-boolean